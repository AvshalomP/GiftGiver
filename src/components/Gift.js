import React, { Component } from 'react';
import styles from '../styles/Gift.module.css';

class Gift extends Component {
    constructor(props){
        super(props);

        this.state = { person: "", present: "" };
    }

    handleChange = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        });
    };

    render(){
        const { handleRemove, gift } = this.props;
        return (

                <div className={styles.container}>
                <form className={styles.formContainer}>
                    <label htmlFor="person">Person</label>
                    <input type="text" className="input-person" name="person"
                           onChange={this.handleChange}/>
                    <label htmlFor="present">Present</label>
                    <input type="text" className="input-present" name="present"
                           onChange={this.handleChange}
                    />
                </form>
                <button id="removeBtn" className={styles.btn}
                        onClick={handleRemove.bind(this, gift.id)}
                >
                    Remove Gift
                </button>
            </div>
        )
    }
}


export default Gift;