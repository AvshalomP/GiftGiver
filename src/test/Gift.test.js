import React from 'react';
import { shallow } from 'enzyme';
import Gift from '../components/Gift';


describe('Gift', () => {
    const gift = shallow(<Gift />);

    it('should render properly', () => {
        expect(gift).toMatchSnapshot();
    });

    it('initializes a person and present in `state`', () => {
        expect(gift.state()).toEqual({ person: "", present: "" });
    });
});

