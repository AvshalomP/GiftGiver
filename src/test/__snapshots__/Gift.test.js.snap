// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift
    gift={
      Object {
        "id": 1,
      }
    }
    handleRemove={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="formContainer"
        >
          <label
            htmlFor="person"
          >
            Person
          </label>
          <input
            className="input-person"
            name="person"
            onChange={[Function]}
            type="text"
          />
          <label
            htmlFor="present"
          >
            Present
          </label>
          <input
            className="input-present"
            name="present"
            onChange={[Function]}
            type="text"
          />
        </form>,
        <button
          className="btn"
          id="removeBtn"
          onClick={[Function]}
        >
          Remove Gift
        </button>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="person"
            >
              Person
            </label>,
            <input
              className="input-person"
              name="person"
              onChange={[Function]}
              type="text"
            />,
            <label
              htmlFor="present"
            >
              Present
            </label>,
            <input
              className="input-present"
              name="present"
              onChange={[Function]}
              type="text"
            />,
          ],
          "className": "formContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Person",
              "htmlFor": "person",
            },
            "ref": null,
            "rendered": "Person",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-person",
              "name": "person",
              "onChange": [Function],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Present",
              "htmlFor": "present",
            },
            "ref": null,
            "rendered": "Present",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-present",
              "name": "present",
              "onChange": [Function],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove Gift",
          "className": "btn",
          "id": "removeBtn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Gift",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="formContainer"
          >
            <label
              htmlFor="person"
            >
              Person
            </label>
            <input
              className="input-person"
              name="person"
              onChange={[Function]}
              type="text"
            />
            <label
              htmlFor="present"
            >
              Present
            </label>
            <input
              className="input-present"
              name="present"
              onChange={[Function]}
              type="text"
            />
          </form>,
          <button
            className="btn"
            id="removeBtn"
            onClick={[Function]}
          >
            Remove Gift
          </button>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="person"
              >
                Person
              </label>,
              <input
                className="input-person"
                name="person"
                onChange={[Function]}
                type="text"
              />,
              <label
                htmlFor="present"
              >
                Present
              </label>,
              <input
                className="input-present"
                name="present"
                onChange={[Function]}
                type="text"
              />,
            ],
            "className": "formContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Person",
                "htmlFor": "person",
              },
              "ref": null,
              "rendered": "Person",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-person",
                "name": "person",
                "onChange": [Function],
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Present",
                "htmlFor": "present",
              },
              "ref": null,
              "rendered": "Present",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-present",
                "name": "present",
                "onChange": [Function],
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove Gift",
            "className": "btn",
            "id": "removeBtn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Gift",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
